DataFormats/Detectors/Common/src/AlignParam.cxx

```diff
@@ -314,7 +328,26 @@ bool AlignParam::applyToGeometry() const
   LOG(debug) << "Aligning volume " << symname;
 
   node->Align(ginv);
-
+  std::string pathStr;
+  if (path) {
+    pathStr.append(path);
+    if (getDetFromSymName(getSymName()) == "ITS" && (pathStr.find("ITSUChip") != std::string::npos) && usePW) {
```

Comment:
from a framework point of view, this looks is a bit unfortunate. We go from something generic to something that contains ITS specific code. Could we think of alternatives, in which we call detector spefific routines adjustments (callback, virtual function, etc.)?

Reply:
in the new implementation the volumes are added to the parallel world in a second step after the alignment. As in the other comment, the ITS code is now handling the filling of ITS volumes